mopidy/stream/actor.py

```diff
@@ -55,3 +62,78 @@ def lookup(self, uri):
             track = Track(uri=uri)
 
         return [track]
+
+
+class StreamPlaybackProvider(backend.PlaybackProvider):
+
+    def __init__(self, audio, backend, config):
+        super(StreamPlaybackProvider, self).__init__(audio, backend)
+        self._config = config
+        self._scanner = backend._scanner
+
+    def translate_uri(self, uri):
+        try:
+            scan_result = self._scanner.scan(uri)
+        except exceptions.ScannerError as e:
+            logger.warning(
+                'Problem scanning URI %s: %s', uri, e)
+            return None
+
+        if not (scan_result.mime.startswith('text/') or
+                scan_result.mime.startswith('application/')):
+            return uri
+
+        content = self._download(uri)
+        if content is None:
+            return None
+
+        tracks = list(playlists.parse(content))
+        if tracks:
+            # TODO Test streams and return first that seems to be playable
+            return tracks[0]
+
+    def _download(self, uri):
+        timeout = self._config['stream']['timeout'] / 1000.0
+
+        session = get_requests_session(
+            proxy_config=self._config['proxy'],
+            user_agent='%s/%s' % (
+                stream.Extension.dist_name, stream.Extension.version))
+
+        try:
+            response = session.get(
+                uri, stream=True, timeout=timeout)
+        except requests.exceptions.Timeout:
+            logger.warning(
+                'Download of stream playlist (%s) failed due to connection '
+                'timeout after %.3fs', uri, timeout)
+            return None
+
+        deadline = time.time() + timeout
+        content = b''
```

Comment:
Can never remember if it is recommended to continue building things like this or to use `io.Bytes`?